pycqed/measurement/detector_functions.py

```diff
@@ -965,17 +965,20 @@ def __init__(self, HS, threshold=1.75, trigger_separation=20e-6,
         self.last_frequency = 0.
         self.threshold = threshold
         self.last = 1.
-        self.trigger_separation = trigger_separation
-        self.demod_mode = demod_mode
+        HS.trigger_separation(trigger_separation)
+        if 'double' in demod_mode:
+            HS.single_sideband_demod(False)
```

Comment:
I would refer to `self.HS` instead of `HS` to ensure the same variable is used throughout the object.